contracts/tests/EASGatekeeper.test.ts

```diff
@@ -0,0 +1,188 @@
+      expect(easGatekeeper).to.not.eq(undefined);
+    });
+
+    it("should fail to deploy when the eas contract address is not valid", async () => {
+      await expect(
+        deployContract("EASGatekeeper", signer, true, ZeroAddress, trustedAttester, toBeArray(schema)),
+      ).to.be.revertedWithCustomError(easGatekeeper, "ZeroAddress");
+    });
+
+    it("should fail to deploy when the attester address is not valid", async () => {
+      await expect(
+        deployContract("EASGatekeeper", signer, true, easAddress, ZeroAddress, toBeArray(schema)),
+      ).to.be.revertedWithCustomError(easGatekeeper, "ZeroAddress");
+    });
+  });
+
+  describe("EASGatekeeper", () => {
+    let maciContract: MACI;
+
+    before(async () => {
+      const r = await deployTestContracts(initialVoiceCreditBalance, STATE_TREE_DEPTH, signer, true, easGatekeeper);
+
+      maciContract = r.maciContract;
+    });
+
+    it("sets MACI instance correctly", async () => {
+      const maciAddress = await maciContract.getAddress();
+      const tx = await easGatekeeper.setMaciInstance(maciAddress);
+      await tx.wait();
+
+      expect(await easGatekeeper.maci()).to.eq(maciAddress);
+    });
+
+    it("should fail to set MACI instance when the caller is not the owner", async () => {
+      const [, secondSigner] = await getSigners();
+      await expect(easGatekeeper.connect(secondSigner).setMaciInstance(await signer.getAddress())).to.be.revertedWith(
+        "Ownable: caller is not the owner",
+      );
+    });
+
+    it("should fail to set MACI instance when the MACI instance is not valid", async () => {
+      await expect(easGatekeeper.setMaciInstance(ZeroAddress)).to.be.revertedWithCustomError(
+        easGatekeeper,
+        "ZeroAddress",
+      );
+    });
+
+    it("should throw when the attestation is not owned by the caller (mocking maci.signUp call)", async () => {
+      await easGatekeeper.setMaciInstance(await signer.getAddress());
+
+      await expect(
+        easGatekeeper.register(await signer.getAddress(), toBeArray(attestation)),
+      ).to.be.revertedWithCustomError(easGatekeeper, "NotYourAttestation");
+    });
+
+    it("should throw when the attestation has been revoked", async () => {
+      await expect(
+        easGatekeeper.register(await signer.getAddress(), toBeArray(revokedAttestation)),
+      ).to.be.revertedWithCustomError(easGatekeeper, "AttestationRevoked");
+    });
+
+    it("should throw when the attestation schema is not the one expected by the gatekeeper", async () => {
+      await easGatekeeper.setMaciInstance(await signer.getAddress());
+      await expect(
+        easGatekeeper.register(await signer.getAddress(), toBeArray(wrongAttestation)),
+      ).to.be.revertedWithCustomError(easGatekeeper, "InvalidSchema");
+    });
+
+    it("should throw when the attestation is not signed by the attestation owner", async () => {
+      await easGatekeeper.setMaciInstance(await signer.getAddress());
```

Comment:
Maybe we can move `firstSigner = await signer.getAddress()` in before hook and just replace all the places where we use it.